File: src/main/java/com/belleileperfumee/belle_ile_parfumee/entity/Account.java
package com.belleileperfumee.belle_ile_parfumee.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "account")
@Data
public class Account {

    @Id
    @Column(length = 255)
    private String email;

    @Column(length = 255, nullable = false)
    private String password;

    @Column(length = 50, nullable = false)
    private String role;
}

File: frontend/src/types/Account.ts
export interface Account {
    email: string;
    role: 'CLIENT' | 'ADMIN';
}

export interface LoginRequest {
    email: string;
    password: string;
}

export interface LoginResponse {
    email: string;
    role: 'CLIENT' | 'ADMIN';
    token: string;
}

export interface RegisterRequest {
    email: string;
    password: string;
}